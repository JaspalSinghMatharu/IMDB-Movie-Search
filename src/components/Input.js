import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    /** value of the input feild */
    value: PropTypes.string,
    /** Placeholder text for input feild */
    placeholder: PropTypes.string,
    /** Label for the input feild */
    label: PropTypes.string,
    /** id for the input */
    id: PropTypes.string,
    /** A function to be called on input change */
    handleChange: PropTypes.func,
}

const defaultProps = {
    value: '',
    placeholder: 'Select',
    label: "",
}

const Input = (props) => {
    const {label, id, value, placeholder, handleChange} = props;
    return (
        <div className="form-item">
            {label && <label htmlFor={id} className="form-label">{label}</label>}
            <input className="form-input" placeholder={placeholder} value={value} id={id} onChange={handleChange} />
        </div>
    )
}

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;

export default Input;