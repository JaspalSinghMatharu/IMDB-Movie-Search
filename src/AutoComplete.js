import React, { useEffect, useState, useCallback } from 'react';
import debounce from 'lodash/debounce';
import Input from './components/Input';

const AutoComplete = (props) => {
    const [search, setSearch] = useState('');
    const [movies, setMovies] = useState([]);
    const [selectedMovie, setSelectedMovie] = useState(null);


    useEffect(() => {
        console.log(selectedMovie)
    }, [selectedMovie]) 

    const handleMovieSelection = (e) => {
        let movieSelected;
        let movieId = e.target.getAttribute('data-movieId');
        if(movieId) {
            movieSelected = movies.filter((movie) => movie.imdbID === movieId);
        }
        setSelectedMovie(movieSelected[0]);
        setMovies([]);
    }

    const getMovies = (searchText) => {
        let url = `https://www.omdbapi.com/?s=${searchText}&apikey=50314ef2`
        fetch(url).then((data) => data.json()).then((data) => {
            if(data.Response === 'True') {
                let filteredSearchData = data.Search.splice(0,5);
                setMovies(filteredSearchData);
            }
            else {
                alert(data.Error)
            }
        }).catch((e) => {
            console.log(e)
        })
    }

    const debouncedGetMovies = useCallback(debounce((param) => {getMovies(param)}, 300), [])

    const handleMovieSearch = (e) => {
        let inputValue = e.target.value;
        setSearch(inputValue);
        if (inputValue.length > 3) {
            debouncedGetMovies(inputValue);
        }
    }

    return (
        <>
            <div className="autocomplete-wrapper">
                <Input value={search} label="Enter movie name" placeholder="Type to search" id="autoComplete" handleChange={handleMovieSearch} />
                {movies &&
                    <ul className="search-results-list">
                        {movies.map((item) => {
                            return (
                                <li data-movieId={item.imdbID} key={item.imdbID} className="search-result__item" onClick={handleMovieSelection}>{item.Title}</li>
                            )
                        })}
                    </ul>
                }
            </div>
            {selectedMovie && 
                <div className="movie-card-wrapper">
                    <div className="movie-card">
                        <img src={selectedMovie.Poster} alt={selectedMovie.Title} />
                        <div className="movie-detail">
                            <p className="movie-title">Movie Name : <span>{selectedMovie.Title}</span></p>
                            <span>Released Year: {selectedMovie.Year}</span>
                        </div>
                    </div>
                </div>
            }
        </>

    )
}

export default AutoComplete