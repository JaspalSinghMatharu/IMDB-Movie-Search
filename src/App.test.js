import { render, screen } from '@testing-library/react';
import AutoComplete from './AutoComplete';
import Input from './components/Input';

test('render Input component', () => {
  const {wrapper} = render(<Input />);
  expect(wrapper).toMatchSnapshot();
});


test('render AutoComplete component', () => {
  const {wrapper} = render(<AutoComplete />);
  expect(wrapper).toMatchSnapshot();
});
