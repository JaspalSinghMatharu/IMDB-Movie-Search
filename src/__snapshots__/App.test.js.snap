// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AutoComplete component 1`] = `undefined`;

exports[`render Input component 1`] = `undefined`;

exports[`render Input component with label 1`] = `undefined`;

exports[`renders learn react link 1`] = `undefined`;
